django_mongodb/compiler.py

```diff
@@ -33,11 +35,21 @@ def execute_sql(
         except EmptyResultSet:
             return iter([]) if result_type == MULTI else None
 
-        return (
-            (self._make_result(row, columns) for row in query.fetch())
-            if result_type == MULTI
-            else self._make_result(next(query.fetch()), columns)
-        )
+        cursor = query.get_cursor()
+        if result_type == SINGLE:
+            try:
+                obj = cursor.next()
+            except StopIteration:
+                return None  # No result
+            else:
+                return self._make_result(obj, columns)
+        # result_type is MULTI
+        cursor.batch_size(chunk_size)
+        result = self.cursor_iter(cursor, chunk_size, columns)
+        if not chunked_fetch:
+            # If using non-chunked reads, read data into memory.
+            return list(result)
+        return result
 
     def results_iter(
```

Comment:
Sorry if this is a dumb question. How is this getting called in relation to `_make_result` or `cursor_iter`

Reply:
It's called afterward, except when `results is None` and this method invokes `execute_sql()` itself. You can look at `ModelIterable.__iter__()` in [django/db/models/query.py](https://github.com/django/django/blob/main/django/db/models/query.py) and see `compiler.execute_sql` and `compiler.results_iter`